path_role.go

```diff
@@ -0,0 +1,433 @@
+		}
+
+		if ttlVal < 0 {
+			return logical.ErrorResponse("ttl cannot be negative"), nil
+		}
+
+		roleEntry.TTL = ttlVal
+	} else {
+		roleEntry.TTL = DefaultTTLSeconds
+	}
+
+	if add_policy_list, ok := data.GetOk("add_policy_list"); ok {
+		roleEntry.PolicyList = add_policy_list.([]string)
+	}
+
+	if err := b.nonLockedSetOCIRole(ctx, req.Storage, roleName, roleEntry); err != nil {
+		return nil, err
+	}
+
+	return &resp, nil
+}
+
+// Update if it already exists
+func (b *backend) pathRoleUpdate(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+
+	roleName := strings.ToLower(data.Get("role").(string))
+	if roleName == "" {
+		return logical.ErrorResponse("missing role"), nil
+	}
+
+	b.roleMutex.Lock()
+	defer b.roleMutex.Unlock()
+
+	roleEntry, err := b.nonLockedOCIRole(ctx, req.Storage, roleName)
+	if err != nil {
+		return nil, err
+	}
+
+	if roleEntry == nil {
+		return logical.ErrorResponse("The specified role does not exist"), nil
+	} else {
+		roleEntry.Version = roleEntry.Version + 1
+	}
+
+	if description, ok := data.GetOk("description"); ok {
+		roleEntry.Description = description.(string)
+	}
+
+	//Add and Remove the OCIDs
+	ocidMap := sliceToMap(roleEntry.OcidList)
+
+	if add_ocid_list, ok := data.GetOk("add_ocid_list"); ok {
+		addOcidSlice := add_ocid_list.([]string)
+		ocidMap = addSliceToMap(addOcidSlice, ocidMap)
+	}
+
+	if remove_ocid_list, ok := data.GetOk("remove_ocid_list"); ok {
+		removeOcidSlice := remove_ocid_list.([]string)
+		ocidMap = removeSliceFromMap(removeOcidSlice, ocidMap)
+	}
+
+	roleEntry.OcidList = mapToSlice(ocidMap)
+
+	if len(roleEntry.OcidList) > MaxOCIDsPerRole {
+		return logical.ErrorResponse("Number of OCIDs for this role exceeds the limit"), nil
+	}
+
+	var resp logical.Response
+
+	ttl, ok := data.GetOk("ttl")
+	if ok {
+		ttlVal := ttl.(int)
+
+		if ttlVal > DefaultTTLSeconds {
+			return logical.ErrorResponse(fmt.Sprintf("Given ttl of %d seconds should be lesser than %d seconds;", ttlVal, DefaultTTLSeconds)), nil
+		}
+
+		if ttlVal < 0 {
+			return logical.ErrorResponse("ttl cannot be negative"), nil
+		}
+
+		roleEntry.TTL = ttlVal
+	}
+
+	//Add and Remove the Policies
+	policyMap := sliceToMap(roleEntry.PolicyList)
+
+	if add_policy_list, ok := data.GetOk("add_policy_list"); ok {
+		addPolicySlice := add_policy_list.([]string)
+		policyMap = addSliceToMap(addPolicySlice, policyMap)
+	}
+
+	if remove_policy_list, ok := data.GetOk("remove_policy_list"); ok {
+		removePolicySlice := remove_policy_list.([]string)
+		policyMap = removeSliceFromMap(removePolicySlice, policyMap)
+	}
+
+	roleEntry.PolicyList = mapToSlice(policyMap)
+
+	if err := b.nonLockedSetOCIRole(ctx, req.Storage, roleName, roleEntry); err != nil {
+		return nil, err
+	}
+
+	return &resp, nil
+}
+
+// Struct to hold the information associated with an OCI role
+type OCIRoleEntry struct {
```

Comment:
Add `tokenutil.TokenParams` here and remove `PolicyList`. This will automatically bring in a set of common role parameters (including policy).  [Example](https://github.com/hashicorp/vault/blob/50704e612cb9ddc110f62dad6beefedf73c6e4ed/builtin/credential/okta/path_config.go#L285)

Reply:
Thanks, have changed the code here and in other places to utilize tokenutil.TokenParams